src/components/Card/TimeSelectorCard.tsx

```diff
@@ -12,37 +13,136 @@ export const TimeSelectorCard: FC<ICardProps> = ({ name }) => {
-                return <div key={`${index}-${name}`}>
-                    <span>{moment(startTime, "h:mm").format('LT')}</span> - <span>{moment(endTime, "h:mm").format('LT')}</span> 
+                const { startTime, endTime } = time || {}
+                return time && <div key={`${index}-${name}`}>
+                    <span>{startTime}</span> - <span>{endTime}</span> 
                     <span onClick={() => deleteTimeSlot(index)}>Delete</span>
                 </div>
             })
         )
     }
 
     const deleteTimeSlot = (index: number) => {
+        const endTimeIndex = timestamp[index].endTimeIndex
+
+        //restore delete timestamps
+        for(let i=index; i< endTimeIndex; i++) {
+            startTimestamp[i] = convertIntegerToTime(i)
+        }
+
+        for(let i=0; i< endTimestamp.length; i++) {
+            endTimestamp[i] = convertIntegerToTime(i)
+        }
+        setStartTimestamp([...startTimestamp])
+        setEndTimestamp([...endTimestamp])
+
+        //delete element from timestamp
         timestamp.splice(index, 1)
         setTimestamp([...timestamp])
     }
 
+    const addTime = () => {
+
+        const startTimeIn24Format = convertTimeIn24Format(startTime)
+        const endTimeIn24Format = convertTimeIn24Format(endTime)
+
+        const startTimeIndex = parseInt(startTimeIn24Format)
+        const endTimeIndex = parseInt(endTimeIn24Format)
+
+        //remove start timestamps which has been allocated
+        if(startTimeIndex > endTimeIndex) {
+            for (let i = 0; i < endTimeIndex; i++)
+                startTimestamp[i] = ''
+
+            for (let i = startTimeIndex; i < startTimestamp.length; i++)
+                startTimestamp[i] = ''
+
+            setStartTimestamp([...startTimestamp])
+        } else {
+            for (let i = startTimeIndex; i < endTimeIndex; i++)
+                startTimestamp[i] = ''
+
+            setStartTimestamp([...startTimestamp])
+        }
+        //restore endTimestamp array
+        for(let i=0; i<endTimestamp.length; i++)
+            endTimestamp[i] = convertIntegerToTime(i)
+
+        timestamp[startTimeIndex] = {startTime, endTime, endTimeIndex}
+    }
+
+    const setStartTimeAndDropdownValues = (startTime: string) => {
+        setStartTime(startTime)
+
+        const startTimeIn24Format = convertTimeIn24Format(startTime)
+        const startTimeIndex = parseInt(startTimeIn24Format)
+
+        if(timestamp.length >= 1) {
+            for (let time of timestamp) {
+                const {startTime, endTime} = time || {}
+                const startTimeOfTimestampIn24Format = convertTimeIn24Format(startTime)
+                const endTimeOfTimestampIn24Format = convertTimeIn24Format(endTime)
+
+                const startTimeIndexOfTimestamp = parseInt(startTimeOfTimestampIn24Format)
+                const endTimeIndexOfTimestamp = parseInt(endTimeOfTimestampIn24Format)
+
+                if(time) {
```

Comment:
i think if we just move this `if` check above right after the for loop then we can avoid all these calculations which become useless if the time is null or undefined

Reply:
good idea. Thanks